apps/frontend/src/components/views/PurchaseCredits/CreditCurrentPrice.tsx

```diff
@@ -0,0 +1,22 @@
+import { usePrices } from '../../../hooks/usePrices';
+
+const MB = 10 ** 6;
+
+export const CreditCurrentPrice = () => {
+  const { shannonsPerByte, formatCreditsAsAi3 } = usePrices();
+
+  return (
+    <div className='rounded-xl bg-primary p-6 text-primary-foreground'>
+      <div className='flex flex-col items-start justify-between gap-4 md:flex-row md:items-center'>
+        <div>
+          <div className='text-sm opacity-90'>Current Price</div>
+          <div className='text-3xl font-bold'>
+            {typeof shannonsPerByte === 'number'
+              ? `${formatCreditsAsAi3(MB).toFixed(2)} AI3 per MB`
```

Comment:
I've read the code, but I'm not sure I understand the rounding strategy.

I think it should be something like:
- when the user is buying, round up the price and megabytes, and round down the credits they'll get, so they always have enough to cover any hidden fractional upload size
- for amounts the user holds, round down the credits they have, so they don't try and upload beyond their actual capacity

Reply:
Makes sense your approach, I didn't implemented that sensitive. The `Number.toFixed` method has Round as its implicit rounding strategy. So, basically we have to explicitly truncate the amounts in two cases:

- Converting AI3 into credits
- Showing user's balance

I'll add this two

Reply:
Updated